pkg/server/apiHandler.go

```diff
@@ -113,16 +117,25 @@ func (h *apiHandler) config(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, http.StatusText(http.StatusUnauthorized), http.StatusUnauthorized)
 		return
 	}
-	config, err := h.manager.GetConfig(*u)
+	msg, err := h.manager.GetConfig(*u)
 	if err != nil {
 		log.Printf("error getting device config: %v", err)
 		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
 		return
 	}
-	out, err := proto.Marshal(config)
+	configResp := &config.ConfigResponse{}
+	blob, err := json.Marshal(msg)
+	if err != nil {
+		log.Printf("error marshaling config to json: %v", err)
+	}
+	checksum := fmt.Sprintf("%X", sha256.Sum256(blob))
+	configResp.Config = msg
```

Comment:
This should only be included if the checksum is different than what was in the configRequest.